### PR TITLE
TD-1116 User feedback mechanism.

### PR DESCRIPTION
**Pull request by martinfoster-hee**
### JIRA link
[TD-1116](https://hee-tis.atlassian.net/jira/software/c/projects/TD/boards/165?modal=detail&selectedIssue=TD-1116)

### Description
Implemented guest and user feedback header bar and pages.

Added new db table UserFeedback.
Updated related GDS.MultiPageFormData nuget packet (v1.0.4).
Updated the two root _layout.cshtml views to show new user feedback header.
Added controller and views for two user journeys: Guest and LoggedIn User.

### Screenshots
Screenshots below.

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [ ] Manually tested my work with and without JavaScript
- [x] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

a
b
c
d
e

selfassessmentheader

[TD-1116]: https://hee-tis.atlassian.net/browse/TD-1116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by kevwhitt-hee** [comment]
This is looking very good, indeed Martin. Building and running it, I saw just one problem with the label of the "Back to" button (it did redirect me back to where I came from - the Learning Portal -> Current Activities view):

image

**Review by rshrirohit — CHANGES_REQUESTED** [review]

File: DigitalLearningSolutions.Web/Views/LearningPortal/SelfAssessments/_Layout.cshtml [hunk 1]
```diff
@@ -64,21 +66,41 @@
...
             </div>
+            <div class="nhsuk-header__service-name">
+              <span class="nhsuk-header__transactional-service-name--link">@ViewData["SelfAssessmentTitle"]</span>
+              <sup class="header-beta">Beta</sup>
+            </div>
+          </div>
+
         </header>
+
+        @if ((bool?)ViewData[LayoutViewDataKeys.DoNotDisplayUserFeedbackBar] != true)
```
[the remark this answers is not in the document]
Comment (martinfoster-hee) [reply DigitalLearningSolutions.Web/Views/LearningPortal/SelfAssessments/_Layout.cshtml:86]: Good call.

**Comment by martinfoster-hee** [comment]
> caching

Ah, good idea!

File: DigitalLearningSolutions.Web/Views/LearningPortal/SelfAssessments/_Layout.cshtml (hunk 1 above)
[the remark this answers is not in the document]
Comment (martinfoster-hee) [reply DigitalLearningSolutions.Web/Views/LearningPortal/SelfAssessments/_Layout.cshtml]: Done.

**Pull request merged**